packages/schema/src/config/common.ts

```diff
@@ -361,7 +361,15 @@ export default defineUntypedSchema({
    * @type {Array<string | RegExp>}
    */
   watch: {
-    $resolve: val => [].concat(val).filter((b: unknown) => typeof b === 'string' || b instanceof RegExp)
+    $resolve: async (val, get) => {
+      let spaLoadingTemplate = await get('spaLoadingTemplate')
+      if (typeof spaLoadingTemplate !== 'string') {
+        spaLoadingTemplate = 'app/spa-loading-template.html'
+      }
+      return [
+        relative(await get('rootDir'), resolve(await get('srcDir'), spaLoadingTemplate))
```

Comment:
For future maintainability, I would create an array `watchFiles` and push to it (instead of directly contact edge case + filter